pkg/controller/job/controller.go

```diff
@@ -300,11 +301,15 @@ func (jm *JobController) syncJob(key string) error {
 	}
 	job := *obj.(*experimental.Job)
 	if !jm.podStoreSynced() {
-		// Sleep so we give the pod reflector goroutine a chance to run.
-		time.Sleep(replicationcontroller.PodStoreSyncedPollPeriod)
-		glog.V(4).Infof("Waiting for pods controller to sync, requeuing job %v", job.Name)
-		jm.enqueueController(&job)
-		return nil
+		// wait.ConditionFunc contract includes an error, so wrap and return nil
+		pollFn := func() (bool, error) {
+			return jm.podStoreSynced(), nil
+		}
+		if err := wait.Poll(replicationcontroller.PodStoreSyncedPollPeriod, replicationcontroller.PodStoreSyncedPollPeriod*10, pollFn); err != nil {
```

Comment:
You're only extending the wait time here, how this should help? Does this applies to RC as well, as this piece of code is pure copy from [here](https://github.com/kubernetes/kubernetes/blob/master/pkg/controller/replication/replication_controller.go#L418-L424)?